### ISSUE TITLE
Tree structure in folder view

### ISSUE THREAD

**Issue by pieter-groeneweg** [Rello/audioplayer#251]
In current app all folders are shown in Alphanumeric order. Subfolders are listed in the same view.
This is a bit odd (i have a discography of Jamiroquay which now is all over the place... and of course in the main folder..

Wouldn't it be nice if the folder structure would be preserved in that view?

**Comment by Rello** [comment]
Hello,
thank you for your request. but this is not planned right now.
it would involve a lot of recursive loops to build up this hierarchical relationship. as soon as we reach many folders, the user would wait possibly several seconds for the reply. we also have the power users with several 1000 titles in scope.

in your case, the Album or Artist view might be the better place I would guess.

**Comment by nilsgoldbeck** [comment]
I would have liked to see this feature added, too. I don't reall understand the performance argument against it. I would have thought that a tree structure would be more efficient than a flat list of folders. Too bad, I will stick to listening offline with Foobar then until I find an web app that offers this simple feature.

**Comment by pieter-groeneweg** [comment]
..I agree.. I can't see why it is harder to have a folder tree than to read and transform a huge load of file details such as album, artist, cover images, year and on. 
And then there is this simple fact the folder tree is already in Nextcloud files view... it only needs displayed in the left of the player..

**Comment by Rello** [comment]
Hello,
here is some background: you have all titles scanned in your library. the `_tracks` table is the source for all details on track level (take e.g. 500 files). Every file has a `folder_id` stored in the table.
Getting all distinct `folder_ids` (take 70 in this case) is one `SELECT` and counting the number of tracks for each is a simple loop. 
thats it.

what you would require is to loop the list of `folder_ids` and get the `parent_id` via the OPC NODES class (70 reads on the filecache). Now for every `parent_id` do another loop; and another one; another one, depending on the levels of your library. In iTunes at least 3 levels.

and with this amount of arrays transferred via AJAX to jQuery, do many other fancy loops to create the required frontend output
https://docs.nextcloud.com/server/12/developer_manual/design/navigation.html#collapsible-entry

**long story short:** a lot of work and performance implications. We are currently prioritising other tasks.
But if you have a suggestion, we are always open to pull requests. AP is always searching for supporters
